commands/azureCommands/delete-image.ts

```diff
@@ -13,38 +13,69 @@ import { AzureImage } from "../../utils/Azure/models/image";
 import { Repository } from "../../utils/Azure/models/repository";
 import * as quickPicks from '../utils/quick-pick-azure';
 
+/** Function to untag an Azure hosted image
+ * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ */
+export async function untagAzureImage(context?: AzureImageTagNode): Promise<void> {
+    removeImage(context, true);
+}
+
 /** Function to delete an Azure hosted image
  * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
  */
 export async function deleteAzureImage(context?: AzureImageTagNode): Promise<void> {
+    removeImage(context, false);
+}
+
+/** Function to delete an Azure hosted image
+ * @param context : if called through right click on AzureImageNode, the node object will be passed in. See azureRegistryNodes.ts for more info
+ * @param untag : if true deletes the image tag, otherwise removes digest and all other tags associeted to the image selected.
+ */
+async function removeImage(context: AzureImageTagNode, untag: boolean): Promise<boolean> {
+    let action: string = (untag) ? "untag" : "delete";
     let registry: Registry;
-    let repoName: string;
-    let tag: string;
+    let repo: Repository;
+    let image: AzureImage;
 
     if (!context) {
         registry = await quickPicks.quickPickACRRegistry();
-        const repository: Repository = await quickPicks.quickPickACRRepository(registry, 'Select the repository of the image you want to delete');
-        repoName = repository.name;
-        const image: AzureImage = await quickPicks.quickPickACRImage(repository, 'Select the image you want to delete');
-        tag = image.tag;
+        repo = await quickPicks.quickPickACRRepository(registry, `Select the repository of the image you want to ${action}`);
+        image = await quickPicks.quickPickACRImage(repo, `Select the image you want to ${action}`);
 
     } else {
         registry = context.registry;
         let wholeName: string[] = context.label.split(':');
```

Comment:
Does label always follow the `repository:tag` format? Just want to make sure `latest` tag is never omitted in label.

Reply:
This is assumption it's all over the project if this is not the case we should fix it. However, I thought this is the case because the extension doesn't allow you to create an image if you do not follow this format.
If this is an issue I think is better if I make a whole PR to fix all of them.

Reply:
OK. Then keep the code as it is.